### DISCUSSION TITLE
Announcing: TypeBox Form Parser

### DISCUSSION THREAD

**Discussion by jtlapp** [sinclairzx81/typebox#534] [Show and tell]
This library interprets HTTP form data and query parameters as objects defined by TypeBox schemas. Given a TypeBox schema and either an instance of [FormData](https://developer.mozilla.org/en-US/docs/Web/API/FormData) or [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams), the library parses the data as an instance of the schema to the degree that it is able, without also validating the data. To validate the resulting parsed data, pass it to TypeBox's check function or errors function, or use a [TypeBox validator](https://github.com/jtlapp/typebox-validators).

https://github.com/jtlapp/typebox-form-parser

**Comment by sinclairzx81** [comment]
@jtlapp Hey awesome :) Nice work!

Would you like to have the library included in the TB ecosystem list?

https://github.com/sinclairzx81/typebox#ecosystem

**Reply by jtlapp** [reply]
Sure! Thanks!

**Reply by sinclairzx81** [reply]
@jtlapp Hi, apologies for the delay on this (it's been a busy week). Have just updated the Ecosystem list to include [typebox-form-parser](https://github.com/jtlapp/typebox-form-parser).

This library looks great! Thanks for sharing!
All the best!
S
